src/server/req_delete.c

```diff
@@ -446,26 +479,33 @@ decr_single_subjob_usage(job *parent)
  * @param[in] preq - request structure
  */
 static void
-reorder_deljoblist(struct batch_request *preq)
+init_deljoblist(struct batch_request *preq)
 {
 	int head = -1;
 	int tail;
 	char **jlist = preq->rq_ind.rq_deletejoblist.rq_jobslist;
+	struct batch_reply *preply = &preq->rq_reply;
 	job *pjob;
 	char *temp;
 
 	if (preq->rq_ind.rq_deletejoblist.rq_resume)
 		return;
 
+	set_mail_count(preq);
```

Comment:
yes, it is consistent with other counters such as queued limits which are all applicable at the instance level.

Reply:
also, think about the jobs deleted from a remote server. We need to obey the remote server counter instead of the original server.

Reply:
I'm not talking about the server attribute, I'm talking about passing -Wsuppress_email from qdel itself, which overrides the  server's limit. Thanks for mentioning remote servers, I hadn't thought about that. We need to adhere to the user provided limit for all servers combined, remote or otherwise, which is why i suspect that this was always handled on the client side, not server. So, this change breaks suppress_email even for single server setup with a remote server, not just for multi-server.

What are we gaining by this code change? we save a statserver() call and an additional deljoblist call? that doesn't seem worth it, so I suggest that we revert this change. If you still want to keep it then you'll need to figure out how to enforce user provided mail limit with multiple servers.

Reply:
I see what you are saying. I can decrement the mail count on the go in qdel for multiple servers. And we can do that in an easier and cleaner way.

OTOH, the current code doesn't honour the mail count at the server side for remote servers. It fetches the value from the first server and assumes the same hold true for remote servers. It is a bug. To solve this, we need to fetch the server attribute from each and every server and batch requests for each server for mails and no-mails as IFL accepts only a flag today. As you can imagine, this just makes things complicated.

Counting the merits:
* Qdel do not have to do an extra server query
* Qdel do not have to batch requests for mail and no-mails
* Allows server-side optimizations like delete queued jobs first to work correctly.
* Fixes an existing bug with remote servers.
* More maintainable code, no multiple batching and stats.
* Finally it's the reasonable way to do it. The client passes along the value it has got to the server and let it deal with it than fetching the value from the server, process it, and batch the requests in pieces because the API accepts only a boolean flag.

Let me know what you think? I'll hold off until I hear back.